### PR TITLE
Move to solr 8.7 with auth

### PR DESCRIPTION
**Pull request by pcm32**
This PR adds documentation to some of the rules and moves to Solr 8.7 with authentication.

### CODE REVIEW

**Comment by pcm32** [comment]
This is ready to be merged. Subsequent improvements for species subgroups will be made in later PRs.

File: Snakefile [hunk 1]
```diff
@@ -482,6 +544,13 @@ rule run_bioentities_JSONL_creation:
         """
 
 rule delete_species_bioentities_index:
+    """
+    This step deleted the current contents of bioentities index for the specified species. We want this
```
Comment (pmb59) [inline Snakefile]:
```suggestion
    This step deletes the current contents of bioentities index for the specified species. We want this
```

File: Snakefile [hunk 2]
```diff
@@ -586,12 +672,19 @@ rule analytics_bioentities_mapping:
         {micromamba_env}
 
         # we don't break this into done and failed as we need a single mapping file per chunk
+        # Note that part of the data from the mapping comes from the Solr server, so as an input the fact that bioentites as been loaded is needed.
```
Comment (pmb59) [inline Snakefile]:
```suggestion
        # Note that part of the data from the mapping comes from the Solr server, so as an input the fact that bioentites has been loaded is needed.
```

**Review by pmb59 — APPROVED** [review]
@pcm32 LGTM!

**Comment by pcm32** [comment]
Thanks for the review @pmb59

**Pull request merged**